packages/sanity/src/core/studio/copyPaste/transferValue.ts

```diff
@@ -258,24 +296,64 @@ export async function transferValue({
 
   // Arrays
   if (sourceSchemaTypeAtPath.jsonType === 'array' && targetSchemaTypeAtPath.jsonType === 'array') {
+    // There will be a mismatch between the sourceSchemaTypeAtPath (uses []) vs sourceValueAtPath (returns 'String')
+    // when copying a single array item that is a primitive value. We will do an extra check here to make sure we
+    // allow for this conversion
+    // @todo Refactor sourcePath to allways be the relative or complete path
+    const wrappedSourceValueAtPath =
+      isSourceSinglePrimitiveArrayItem && !Array.isArray(sourceValueAtPath)
+        ? [sourceValueAtPath]
+        : sourceValueAtPath
+
     return collateArrayValue({
-      sourceValue: sourceValueAtPath as unknown[],
+      sourceValue: wrappedSourceValueAtPath,
       targetSchemaType: targetSchemaTypeAtPath as ArraySchemaType,
       targetRootValue,
       targetRootPath,
       errors,
+      options,
       keyGenerator,
     })
   }
 
-  // If this is a primitive source and primitive array target, we need to wrap the source value in an array
-  if (isPrimitiveSourceAndPrimitiveArrayTarget) {
+  // If this is a primitive source and primitive array target OR an object source and array of objects target, we need to wrap the source value in an array
+  if (isPrimitiveSourceAndPrimitiveArrayTarget || isObjectSourceAndArrayOfObjectsTarget) {
+    // Here we check if the source value does not contain a type for some reason, or its the type "object". This happens if you copy a object into a array
+    // Then we need to get the type from the path
+    let objectType
+
+    // Check if it's an object source with array of objects target
+    // and the source value is a typed object with '_type' of 'object',
+    // OR if the source value is not a typed object
+    if (
+      (isObjectSourceAndArrayOfObjectsTarget &&
+        isTypedObject(sourceValueAtPath) &&
+        sourceValueAtPath._type === 'object') ||
+      !isTypedObject(sourceValueAtPath)
+    ) {
+      // In this case, determine the object type from the source root path
+      objectType = getObjectTypeFromPath(sourceRootPath)
```

Comment:
I'm unsure what this part is trying to target. Can you give me an example of a use case? 🙏

Reply:
Example test case is here: https://github.com/sanity-io/sanity/blob/627b03951f260402b92a7c9a20afb67a924e6564/packages/sanity/src/core/studio/copyPaste/__test__/transferValue.test.ts#L771

Reply:
Thank you 🙏